src/executor/seq_scan_executor.cpp

```diff
@@ -219,5 +226,81 @@ bool SeqScanExecutor::DExecute() {
   return false;
 }
 
+// Update Predicate expression
+// this is used in the NLJoin executor
+void SeqScanExecutor::UpdatePredicate(const std::vector<oid_t> &column_ids,
+                                      const std::vector<type::Value> &values) {
+  std::vector<oid_t> predicate_column_ids;
+
+  PL_ASSERT(column_ids.size() <= column_ids_.size());
+
+  // columns_ids is the column id
+  // in the join executor, should
+  // convert to the column id in the
+  // seq scan executor
+  for (auto column_id : column_ids) {
+    predicate_column_ids.push_back(column_ids_[column_id]);
+  }
+
+  expression::AbstractExpression *new_predicate =
+      values.size() != 0 ? ColumnsValuesToExpr(predicate_column_ids, values, 0)
+                         : nullptr;
+
+  // combine with original predicate
+  if (old_predicate_ != nullptr) {
+    expression::AbstractExpression *lexpr = new_predicate,
+                                   *rexpr = old_predicate_->Copy();
+
+    new_predicate = new expression::ConjunctionExpression(
+        ExpressionType::CONJUNCTION_AND, lexpr, rexpr);
+  }
+
+  // Currently a hack that prevent memory leak
+  // we should eventually make prediate_ a unique_ptr
+  new_predicate_.reset(new_predicate);
+  predicate_ = new_predicate;
```

Comment:
This is the Optimizer team's code(which has not yet been merged) which we needed to get our code working.